app/services/checksum_validator.rb

```diff
@@ -87,30 +102,39 @@ def add_cv_result_for_deletions_in_xml(subentity)
     end
   end
 
-  # exploring a solution for #522
-  # for latest sigcat, compare checksums
-  def validate_signature_catalog
-    results = []
-    signature_catalog_entries.each do |sig_cat_entry|
-      calculated_signature = Moab::FileSignature.new.signature_from_file(sig_cat_entry_path(sig_cat_entry))
-      results << sig_cat_entry.signature.eql?(calculated_signature)
+  def validate_signature_catalog_entry(entry)
+    calculated_signature = Moab::FileSignature.new.signature_from_file(signature_catalog_entry_path(entry))
+    unless entry.signature.eql?(calculated_signature)
+      mismatch_error_data = { file_path: signature_catalog_entry_path(entry).to_s, version: entry.version_id }
+      checksum_results.add_result(AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, mismatch_error_data)
     end
-    results
+  rescue Errno::ENOENT
+    absent_from_moab_data = { manifest_file_path: latest_signature_catalog_path,
+                              file_path: signature_catalog_entry_path(entry).to_s }
+    checksum_results.add_result(AuditResults::FILE_NOT_IN_MOAB, absent_from_moab_data)
   end
 
   def moab_storage_object
     Moab::StorageObject.new(druid, druid_path)
   end
 
-  def signature_catalog_entries
-    moab_storage_object.version_list.last.signature_catalog.entries
-  end
-
   def druid_path
     "#{endpoint.storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
   end
 
-  def sig_cat_entry_path(e)
-    Pathname("#{druid_path}/#{Moab::StorageObject.version_dirname(e.version_id)}/data/#{e.group_id}/#{e.path}")
+  def signature_catalog_entry_path(e)
+    Pathname("#{druid_path}/#{Moab::StorageObject.version_dirname(e.version_id)}/#{DATA}/#{e.group_id}/#{e.path}")
+  end
```

Comment:
Moab::StorageObjectVersion has `attr_accessor :version_pathname` which should mean this can be simplified to 

```ruby
latest_moab_version.version_pathname.join(DATA, e.group_id, e.path)
```
which might be a little easier to follow?

Reply:
I really like the idea of making this simpler, but these two approaches don't return the same thing. For example, for druid `bj102hs9687` in `fixture_sr3`,  testing against the `first` entry in the `latest_signature_catalog_entries`: 
```
Pathname("#{druid_path}/#{Moab::StorageObject.version_dirname(e.version_id)}")
=> #<Pathname:spec/fixtures/checksum_root01/moab_storage_trunk/bj/102/hs/9687/bj102hs9687/v0001>
latest_moab_version.version_pathname
=> #<Pathname:spec/fixtures/checksum_root01/moab_storage_trunk/bj/102/hs/9687/bj102hs9687/v0003>
```
The approach you're suggesting will point to the latest version of the moab, but we need the `e.version_id` instead, which tells us which version the entry lives in.  I'd be happy for us to find a better way, though.